extensions/qrcode-generator/src/clipboard.tsx

```diff
@@ -1,40 +1,90 @@
 import { Detail, Clipboard, Toast, showToast } from "@raycast/api";
-import { useEffect, useState } from "react";
-import { toDataURL } from "qrcode";
+import { useEffect, useRef, useState } from "react";
+import { toDataURL, toBuffer } from "qrcode";
+import fse from "fs-extra";
+import { homedir } from "os";
+
+export function buildFileName(path: string, name: string, extension: string) {
+  const directoryExists = fse.existsSync(path + name + "." + extension);
+  if (!directoryExists) {
+    return name + "." + extension;
+  } else {
+    let index = 2;
+    while (directoryExists) {
+      const newName = name + "-" + index + "." + extension;
+      const directoryExists = fse.existsSync(path + newName);
+      if (!directoryExists) {
+        return newName;
+      }
+      index++;
+    }
+    return name + "-" + index + "." + extension;
+  }
+}
 
 export default function Command() {
   const [qrData, setQrData] = useState<string>();
-
+  const [clipboardContent, setClipboardContent] = useState<string>();
+  const hasCopy = useRef(false);
```

Comment:
Is `hasCopy` necessary? The QR code should be generated only once because of the empty `useEffect` anyways, shouldn't it?

Reply:
Did you see this @chenpx976